OAuth2/OAuth2ClientCredentials.swift

```diff
@@ -81,8 +81,15 @@ public class OAuth2ClientCredentials: OAuth2
 		req.HTTPMethod = "POST"
 		req.setValue("application/x-www-form-urlencoded; charset=utf-8", forHTTPHeaderField: "Content-Type")
 		req.setValue("application/json", forHTTPHeaderField: "Accept")
-		req.HTTPBody = "grant_type=client_credentials".dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: true)
-		
+        
+        //check if scope is set
+        if nil != scope{
+            req.HTTPBody = "grant_type=client_credentials&scope=\(scope!)".dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: true)
```

Comment:
Just to be safe, scope should be form-encoded, just use `\(scope.wwwFormURLEncodedString)`.